src/renderer/components/dock/edit-resource/edit-resource-model/call-for-resource/call-for-resource.injectable.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import type { KubeObject } from "../../../../../../common/k8s-api/kube-object";
+import { parseKubeApi } from "../../../../../../common/k8s-api/kube-api-parse";
+import getKubeApiFromPathInjectable from "../../../../../../common/k8s-api/kube-api/get-kube-api-from-path.injectable";
+import type { AsyncResult } from "../../../../../../common/utils/async-result";
+import { getErrorMessage } from "../../../../../../common/utils/get-error-message";
+
+export type CallForResource = (
+  selfLink: string
+) => Promise<AsyncResult<KubeObject | undefined>>;
+
+const callForResourceInjectable = getInjectable({
```

Comment:
Is the whole of this really necessary for unit testing or for the correct behaviour?

Reply:
Both. This is something that I would expect to see in the unit test. `when I do something, calls for resource` 
In the beginning I had it implemented using stores, but changed since that would introduce race-condition when you are refreshing page with tab open. (store might not be loaded yet or not going to be loaded at all, therefore we would have to make sure we load the store here) 

Currently I don't see a place for `ApiManager`, `XStore extends KubeObjectStore` and `XApi extends KubeApi` because all of these can be done a lot simpler. More on this topic in the future if we start doing anything about it.

Reply:
Well what about CRD APIs? Wouldn't the place for ApiManager be to handle all of those things?

As for the behaviour, maybe that behaviour is not correct? Should we really always call for resource?

Reply:
> Well what about CRD APIs? Wouldn't the place for ApiManager be to handle all of those things? 

Not relevant here. We'll figure out when needed.

Reply:
> As for the behaviour, maybe that behaviour is not correct? Should we really always call for resource?

Definitely yes. Editing resource shouldn't have anything to do with how resources are listed. Those are two separate features which **may** share code (or even data) but they should work in isolation. There's no need to couple features together.

Reply:
Benefit of this is that now change in how we list resources cannot broke editing of resources. I'd say that's quite good thing?

Reply:
How is it not relevant? If I want to edit a Custom Resource instance I don't see behaviour to cover that and I also don't see implementation to cover that, plus the user should still be able to do it.

Reply:
We'll that's different thing. I've just added behaviour for testing namespaces, but surely there should be a behaviour for editing custom resources.